teps/0089-spire-support.md

```diff
@@ -0,0 +1,262 @@
+
+### Requirements
+* SPIRE is configured to only issue SVIDs to the Tekton controller via a [k8s Workload Attestor](https://github.com/spiffe/spire/blob/main/doc/plugin_agent_workloadattestor_k8s.md)
+
+### Goals
+
+* Results can be verified as non-falsifiabe
+* A TaskRun yaml can be verified as non-falsifiable
+* Clients are confident that TaskRuns were initiated and monitored by the Tekton controller
+
+### Non-Goals
+* Protect against a malicious cluster admin (this will be a goal for SLSA 4)
+
+
+
+## Proposal
+As mentioned above, the basic design looks like this (this is meant to be high level and still needs to be fleshed out a bit):
+1. Tekton Pipelines receives a TaskRun config, and generates the Pod for it with SPIRE mounted in
+1. The Pod executes, and the entrypointer requests an SVID and signature over the Results
+1. Tekton Pipelines verifies the Results
+1. Meanwhile, Tekton Pipelines has been verifying that the TaskRun hasn't been modified during execution
+1. Tekton Pipelines requests a signature and SVID over the completed TaskRun yaml from SPIRE
+1. Tekton Pipelines stores the SVID and signature as annotations on the TaskRun
+1. Tekton Chains observes the TaskRun, and verifies that SVID and signature against the TaskRun yaml
+1. If verification is successful, Chains proceeds normally. Otherwise, it stops and doesn't sign anything!
+
+
+Then, Chains gets the TaskRun.
+Chains will:
+1. Verify the signature against the SPIRE SVID
+1. If verification fails, Chains will mark the TaskRun as "failed" and move on
+1. Otherwise, continue signing stuff! 
+
+## Implementation Plan
+The plan to achieve non-falsifiable provenance will be implemented in phases.
+
+**Phase 1**
+* Add support for Signed Results with SPIRE (this will primarily involve modifications to the entrypointer image)
+* Add support for Chains verifying Signed Results
+
+**Phase 2**
+* Implement Signed TaskRuns with SPIRE (requires further design)
+* Determine an alternate release process for the SPIRE feature (since the Pipelines Controller will need the SPIRE volume mounted in)
+* Add support for Chains verifying Signed TaskRuns
+
+In parallel with this work, we should:
+* Confirm that this meets the SLSA defintion of "non-falsifiable", which might require a security audit
+
+### Risks and Mitigations
+We'll need to depend on the [github.com/spiffe/go-spiffe](https://github.com/spiffe/go-spiffe) library to interact with the SPIRE agent, request SVIDs and signatures.
+
+
+## Design Details - Signed Results
+
+For now, this design section will focus on Part 1 of the implementation plan: Signed Results.
+This TEP will be updated as we flesh out the design for Signed Taskruns.
+
+### Enabling SPIRE on the Entrypointer Image
+We can add a feature flag `--enable-spire=true` as described in [Customizing the Pipelines Controller behavior](https://github.com/tektoncd/pipeline/blob/main/docs/install.md#customizing-the-pipelines-controller-behavior) as an alpha feature.
```

Comment:
np if we go ahead with this flag but i was wondering, you also mentioned the keyless signing alternative, im wondering if instead of a flag here we could have something like `--enforce-nonfalsifiablity=spire` (or something easier to type...) which could support other options later.

might be over thinking it tho!

Reply:
sure, i think that's a good idea! changed the name of the flag in the TEP :)